src/arrayMethodReduce.js

```diff
@@ -5,7 +5,38 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    if (typeof callback !== 'function') {
+      throw new Error('not a function');
+    }
+
+    if (this.length === 0 && !initialValue) {
+      throw new Error('no initial value and array is empty');
+    }
+
+    let k = 0;
```

Comment:
Try to give variables more precise names it will make the code more understandable.